martin/src/utils/id_resolver.rs

```diff
@@ -57,63 +60,72 @@ impl IdResolver {
+            match names.entry(stanitised_name.clone()) {
                 Entry::Vacant(e) => {
-                    let id = e.key().clone();
+                    if stanitised_name != name {
+                        warn!(
+                            "Source `{name}`{info} was renamed to `{stanitised_name}` as source IDs may only contain alpha-numeric characters or `._-`",
+                            info = info.map_or(String::new(), |v| format!(" ({v})"))
+                        );
+                    }
                     e.insert(unique_name);
-                    return id;
+                    return stanitised_name;
                 }
-                #[allow(clippy::assigning_clones)]
                 Entry::Occupied(e) => {
-                    name = e.key().clone();
                     if e.get() == &unique_name {
-                        return name;
+                        return stanitised_name;
                     }
                 }
             }
         }
-        // name already exists, try it with ".1", ".2", etc. until the value matches
-        // assume that reserved keywords never end in a "dot number", so don't check
+
+        // need to rename => try ".1", ".2", etc. until the value matches
+        // assumes that reserved keywords never end in a "dot number" => no special case for this
         let mut index: i32 = 1;
         let mut new_name = String::new();
         loop {
             new_name.clear();
             write!(&mut new_name, "{name}.{index}").unwrap();
             index = index.checked_add(1).unwrap();
             match names.entry(new_name.clone()) {
+                // found new name
                 Entry::Vacant(e) => {
+                    if is_reserved_name {
+                        warn!(
+                            "Source `{name}`{info} was renamed to `{new_name}` as {name} is a reserved keyword",
+                            info = info.map_or(String::new(), |v| format!(" ({v})"))
+                        );
+                    } else {
+                        warn!(
+                            "Source `{name}`{info} was renamed to `{new_name}` as source IDs must be unique with alpha-numeric characters or `._-`",
+                            info = info.map_or(String::new(), |v| format!(" ({v})"))
+                        );
+                    }
                     e.insert(unique_name);
                     return new_name;
                 }
                 Entry::Occupied(e) => {
                     if e.get() == &unique_name {
+                        // found existing name
                         return new_name;
                     }
                 }
             }
         }
     }
+    /// replace prohibited characters, except underscores, dashes, and dots with dashes.
+    fn sanitise(name: &str) -> String {
+        name.replace(
+            |c: char| !c.is_ascii_alphanumeric() && c != '_' && c != '.' && c != '-',
+            "-",
+        )
```

Comment:
in an ideal world, I would like to trim leading/trailing `-` but given how breaking this might be, lets not do this.